src/snowflake/snowpark/mock/mock_analyzer.py

```diff
@@ -752,6 +754,9 @@ def do_resolve_with_resolved_children(
                 logical_plan,
             )
 
+        if isinstance(logical_plan, Selectable):
```

Comment:
We shouldn't have `Selectable` in our codepath, only `MockSelectable` should be wrapped around by `MockExecutionPlan`